test/memory/MemoryTestPublisher.cpp

```diff
@@ -0,0 +1,393 @@
+// Copyright 2016 Proyectos y Sistemas de Mantenimiento SL (eProsima).
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @file MemoryPublisher.cpp
+ *
+ */
+
+#include "MemoryTestPublisher.h"
+#include "fastrtps/log/Log.h"
+#include "fastrtps/log/Colors.h"
+#include <numeric>
+#include <cmath>
+#include <fstream>
+#include <inttypes.h>
+
+#ifndef _WIN32
+#define localtime_s(X, Y) localtime_r(Y, X)
+#endif
+
+#define TIME_LIMIT_US 10000
+
+using namespace eprosima;
+using namespace eprosima::fastrtps;
+using namespace eprosima::fastrtps::rtps;
+
+
+MemoryTestPublisher::MemoryTestPublisher():
+    mp_participant(nullptr),
+    mp_datapub(nullptr),
+    mp_commandpub(nullptr),
+    mp_commandsub(nullptr),
+    mp_memory(nullptr),
+    n_subscribers(0),
+    n_samples(0),
+    disc_count_(0),
+    comm_count_(0),
+    data_count_(0),
+    m_status(0),
+    n_received(0),
+    m_datapublistener(nullptr),
+    m_commandpublistener(nullptr),
+    m_commandsublistener(nullptr),
+    m_data_size(0)
+{
+    m_datapublistener.mp_up = this;
+    m_commandpublistener.mp_up = this;
+    m_commandsublistener.mp_up = this;
+    m_exportPrefix = "";
+}
+
+MemoryTestPublisher::~MemoryTestPublisher()
+{
+    Domain::removeParticipant(mp_participant);
+}
+
+
+bool MemoryTestPublisher::init(int n_sub, int n_sam, bool reliable, uint32_t pid, bool hostname, bool export_csv,
+        const std::string& export_prefix, const PropertyPolicy& part_property_policy,
+        const PropertyPolicy& property_policy, const std::string& sXMLConfigFile, uint32_t data_size)
+{
+    m_sXMLConfigFile = sXMLConfigFile;
+    n_samples = n_sam;
+    n_subscribers = n_sub;
+    n_export_csv = export_csv;
+    m_exportPrefix = export_prefix;
+    reliable_ = reliable;
+    m_data_size = data_size;
+
+    if (m_sXMLConfigFile.length() > 0)
+    {
```

Comment:
Join code? instead of two blocks, only one. At configuration time try to run `XMLProfileManager::loadXMLFile` if m_sXMLConfigFile.length() > 0 and if it fails exists. In other case set manually the configuration.